### DISCUSSION TITLE
NginxProxyManager / letsencrypt / Wildcard using custom DNS

### DISCUSSION THREAD

**Discussion by AboveNBeyondCode** [NginxProxyManager/nginx-proxy-manager#3353] [Q&A]
Hello,

I have a problem / question.
I am running currently ningx on a linux VM and I want to switch to NginxProxyManager.
Currently I am using a wildcard cert renewing thru certbot using the DNS challenge.
My DNS Server is not in the list of supported DNS servers in the NginxProxyManager UI on the SSL page.

Is there a possibility to use NginxProxyManager with SSL wildcard certs without using one of the predefined DNS server apis?

**Comment by JVKeller** [comment]
I have the same question. I'd like to be able to make a wildcard cert and need to use the DNS Challenge but my host is not in the list. CloudAccess.net

Is there a way to do it with DNS entries like in the cli verison?

**Comment by roswitina** [comment]
I also have the same wish. I want to create a wildcard certificate with my provider hosttech.at, but it is not listed.

Is it perhaps intended to implement the certbot dns multi plugin?

**Comment by slydogsz** [comment]
I want to add my voice to this request. I use dnsexit.com for my dns provider. They have been around for quite some time.
Is there a way or process to get them added to the list, they fully support lets encrypt.

**Comment by JVKeller** [comment]
I figured a way around this.

If you set up a free account on cloudflair, and use then as your name
server, you can then generate the API code on their system and use it in
NPM.

Not sure why it took me so long to realize this. 🙄

**Reply by darkknight7777777** [reply]
How is this a workaround? It's just switching(or adding) dns providers.

**Reply by JVKeller** [reply]
Because it's working around the problem? 5 minutes of setup, and you can generate the API credentials needed to allow NPM to generate the certs.

**Comment by dualizeo** [comment]
Can someone make a tutorial on this?

**Reply by JVKeller** [reply]
Really doesn't need a tutorial? Off the top of my head:

- Just make a Cloud Flare account
- add the website
     - it should copy over all of your records, make sure they look right
     - verify the domain
- change your name servers to CF's
- wait for new settings take effect 
- go to "My Profile"
- go to API
- Use API with NPM

**Comment by slydogsz** [comment]
I would say that this is 100% what I did and it works great with cloudflare. My DNS is still at DNSexit but everything is working great for me.
